package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,503 @@
+        if not hasattr(self._ag, 'bonds'):
+            raise AttributeError('AtomGroup has no attribute bonds')
+        if len(self._ag.fragments) > 1:
+            raise ValueError('AtomGroup has more than one molecule')
+
+        # Determine the root
+        # The initial atom must be a terminal atom
+        terminal_atoms = _sort_atoms_by_mass(\
+            [a for a in self._ag.atoms if len(a.bonds)==1], reverse=True)
+        if (initial_atom is None):
+            # Select the heaviest root atoms from the heaviest terminal atom
+            initial_atom = terminal_atoms[0]
+        elif (not initial_atom in terminal_atoms):
+            raise ValueError('Initial atom is not a terminal atom')
+        # The next atom in the root is bonded to the initial atom
+        # Since the initial atom is a terminal atom, there is only
+        # one bonded atom
+        second_atom = initial_atom.bonded_atoms[0]
+        # The last atom in the root is the heaviest atom
+        # bonded to the second atom
+        # If there are more than three atoms,
+        # then the last atom cannot be a terminal atom.
+        if self._ag.n_atoms != 3:
+            third_atom = _sort_atoms_by_mass(\
+                [a for a in second_atom.bonded_atoms \
+                if (a in self._ag) and (a!=initial_atom) \
+                and (a not in terminal_atoms)], \
+                reverse=True)[0]
+        else:
+            third_atom = _sort_atoms_by_mass(\
+                [a for a in second_atom.bonded_atoms \
+                if (a in self._ag) and (a!=initial_atom)], \
+                reverse=True)[0]
+        self._root = mda.AtomGroup([initial_atom, second_atom, third_atom])
+
+        # Construct a list of torsion angles
+        self._torsions = _find_torsions(self._root, self._ag)
+
+        # Get indices of the root and torsion atoms
+        # in a Cartesian positions array that matches the AtomGroup
+        self._root_XYZ_inds = [(self._ag.indices==a.index).nonzero()[0][0] \
+          for a in self._root]
+        self._torsion_XYZ_inds = [[(self._ag.indices==a.index).nonzero()[0][0] \
+          for a in t] for t in self._torsions]
+
+        # The primary torsion is the first torsion on the list
+        # with the same central atoms
+        prior_atoms = [sorted([a1, a2]) for (a0, a1, a2, a3) in self._torsions]
+        self._primary_torsion_indices = [prior_atoms.index(prior_atoms[n]) \
+          for n in range(len(prior_atoms))]
+        self._unique_primary_torsion_indices = \
+          list(set(self._primary_torsion_indices))
+
+        self._ag1 = mda.AtomGroup([ag[0] for ag in self._torsions])
+        self._ag2 = mda.AtomGroup([ag[1] for ag in self._torsions])
+        self._ag3 = mda.AtomGroup([ag[2] for ag in self._torsions])
+        self._ag4 = mda.AtomGroup([ag[3] for ag in self._torsions])
+
+        if bat_nc is not None:
+            self.load_bat(bat_nc)
+
+    def _prepare(self):
+        self.bat = []
```

Comment:
See comments below why I suggest this
```suggestion
        self._bat = []
```

Reply:
Switched to an numpy array